### ISSUE TITLE
dispatch_get_current_queue is deprecated in iOS 6

### ISSUE THREAD

**Issue by MaxGabriel** [inkling/Subliminal#164]
If you set the deployment target of the Subliminal Project to 6.0+ and then try to build the Subliminal library, there are many deprecation notices for `dispatch_get_current_queue` being deprecated. Based on a quick look, it looks like Subliminal mostly uses that function for the pattern 'if I am not on that queue, dispatch sync to that queue, otherwise just execute normally'. 

I think the proposed workaround for that pattern is to use the `dispatch_queue_set_specific` and `dispatch_get_specific` APIs:

``` objc
@interface ViewController ()
@property (nonatomic, strong) dispatch_queue_t myQueue;
@end

@implementation ViewController

BOOL isMyQueue(void) {
    return dispatch_get_specific(@selector(myQueue)) != NULL;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    NSLog(@"Loading");


    self.myQueue = dispatch_queue_create("myqueue", DISPATCH_QUEUE_SERIAL);
    dispatch_queue_set_specific(self.myQueue, @selector(myQueue), @selector(myQueue), NULL);
    NSLog(@"Is My Queue from main = %i",isMyQueue()); // False

    dispatch_async(self.myQueue, ^{
        NSLog(@"Is My Queue from async = %i",isMyQueue()); // True
    });
}
```

References:

http://stackoverflow.com/a/12806754/1176156
https://devforums.apple.com/thread/162529?start=0&tstart=0

(My implementation is mostly cropped from the second page of the dev forums post, except I used the trick of [using a selector for the `const void *`](http://nshipster.com/associated-objects/))

**Comment by wearhere** [comment]
Nice find! I had previously seen the warning, done some searches, and found only [answers](http://stackoverflow.com/questions/13237417/alternatives-to-dispatch-get-current-queue-for-completion-blocks-in-ios-6) where people were like "why were you using `dispatch_get_current_queue` to begin with." >.< (See also the first page of that dev forum discussion, sheesh.)

It'd be good to pick this up sometime in the near future, certainly before iOS 8. I endorse your approach except for that I prefer

``` objc
static const void *const kIsMyQueueKey = &kIsMyQueueKey;
```

to using selectors as keys. While convenient, selectors don't guarantee that you use the same key across API calls--you could easily mistype a selector, especially if you have the "unknown selector" warning turned off (as Subliminal does, so that the compiler doesn't complain about the use of app hooks).

**Comment by MaxGabriel** [comment]
Yeah jeez Jeff what were you thinking using such a dangerous API like `dispatch_get_current_queue`

**Comment by MaxGabriel** [comment]
Your explanation for not using the selector makes sense; I'll stick to the `x = &x` style.

This doesn't seem like too complex a PR so I'll probably do it.

**Comment by wearhere** [comment]
1.1 so that the users don't see warnings when they install Subliminal.

**Comment by wearhere** [comment]
Closed by #168.
